ls1intum/Artemis
# `Plagiarism checks`: Disable status buttons during confirm and deny action to avoid concurrency issues

### Checklist
#### General


#### Client

### Motivation and Context


Fixes #6529 

### Description

Block both status button when REST call is performed or confirmation pop up is on.

### Steps for Testing

First reproduce the issue on `development`:
1. Create text exercises.
2. Submit two identical submissions for it.
3. Run plagiarism checks.
4. Open plagiarism results.
5. Open Network tab in dev tool and set network throttling to 3G.
6. Confirm the submission and see that buttons were enabled during the REST call.
7. Deny the submission and see that button was enabled during the REST call and pop up on.

Deploy this branch and do steps 4 and 5. Then set up throttling and verify that both buttons are always disabled during the REST call or pop up on.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage




| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| plagiarism-header.component.ts | 100% | ✅                           |
